transport/src/main/java/io/scalecube/transport/TransportEndpoint.java

```diff
@@ -56,31 +164,16 @@ public boolean equals(Object other) {
       return false;
     }
     TransportEndpoint that = (TransportEndpoint) other;
-    return Objects.equals(id, that.id)
-        && Objects.equals(address, that.address);
+    return Objects.equals(id, that.id) && Objects.equals(getSocketAddress(), that.getSocketAddress());
```

Comment:
I think here more logical to compare hostAddress and port rather than transient/computable socketAddress field